pkg/controllers/deprovisioning/types.go

```diff
@@ -119,7 +119,10 @@ func NewCandidate(ctx context.Context, kubeClient client.Client, recorder events
 		pods:         pods,
 	}
 	cn.disruptionCost = disruptionCost(ctx, pods) * cn.lifetimeRemaining(clk)
-	return cn, nil
+	if isCandidateDeprovisionable(cn, recorder, pdbs) {
```

Comment:
Do you think that we could break this up so that we just used the pods and the pdbs directly to do the filtering before we build the candidate? And then we wouldn't have to pass the candidate into the filter function.

Reply:
It makes more sense to me to do the filtering on the "raw node" object rather than against the karpenter-specific object wrapping ("candidate"). I think we have all the data we need for evaluating candidacy then.

I think this is what you're talking about?